### PR TITLE
Add zero, zeros_like, _dimI and _dimV for sparse tensors

### PR DESCRIPTION
**Pull request by ssnl**
This will enable better code for #3137 , #3211 , and future sparse optimizers.

### CODE REVIEW

**Comment by SsnL** [comment]
I'm going to write tests for this, but am not sure where those should go.

**Comment by ezyang** [comment]
If there ever were a case for getting us to stop writing TH/THC/THS/THCS code, and put it in ATen, *this surely is it!* CC @zdevito

File: torch/lib/THS/generic/THSTensorMath.c [hunk 1]
```diff
@@ -15,6 +15,18 @@ void THSTensor_(zero)(THSTensor *self) {
   self->nnz = 0;
 }
 
+void THSTensor_(zeros)(THSTensor *r_, THLongStorage *size)
+{
+  THSTensor_(resize)(r_, size);
```
Comment (apaszke) [inline torch/lib/THS/generic/THSTensorMath.c]: Doesn't sparse resize fix indices or sth? It seems a more expensive call than for dense resize. Maybe zero it beforehand (zero is cheap in sparse)

Comment (SsnL) [reply torch/lib/THS/generic/THSTensorMath.c]: Actually sparse resize doesn't fix indices. It simply sets the sizes. cpu:https://github.com/pytorch/pytorch/blob/master/torch/lib/THS/generic/THSTensor.c#L266 (same for gpu)

**Comment by SsnL** [comment]
@apaszke do the new changes look good to you? :)

File: test/test_sparse.py [hunk 2]
```diff
@@ -545,6 +545,44 @@ def test_sparse_mask(self):
...
+
+    def test_zeros(self):
+        i_shapes = [2, 3, 4]
+        v_shapes = [3, 4, 5, 6]
+        for i_dim in range(1, len(i_shapes) + 1):
+            for v_dim in range(len(v_shapes) + 1):
+                self._test_zeros([2, 3, 4], i_shapes[:i_dim], v_shapes[:v_dim])
+
+    def _test_zeros_like(self, template_shape_i, template_shape_v=None):
+        # import pdb; pdb.set_trace()
```
Comment (apaszke) [inline test/test_sparse.py]: :)

**Review by apaszke — COMMENTED** [review]
LGTM. Just remove the debug code

File: test/test_sparse.py (hunk 2 above)
Comment (SsnL) [reply test/test_sparse.py]: Haha sorry my bad :P

**Comment by SsnL** [comment]
Can we have this merged please :) I have 2 PRs depending on it.

**Pull request merged**